File: demo/src/main/java/Borala/app/demo/Model/Event.java
package Borala.app.demo.Model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "eventos")
@Getter
@Setter
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String detail;

    @Column(nullable = false)
    private LocalDateTime startDate;

    @Column(nullable = false)
    private LocalDateTime endDate;

    @Column(columnDefinition = "TEXT")
    private String details;
}

File: lib/models/event.dart
class Event {
  final int id;
  final String name;
  final String details;
  final String startDate;
  final String endDate;

  Event({
    required this.id,
    required this.name,
    required this.details,
    required this.startDate,
    required this.endDate,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'],
      name: json['name'],
      details: json['detail'],
      startDate: json['startDate'],
      endDate: json['endDate'],
    );
  }
}
